src/Toml/BiMap.hs

```diff
@@ -189,38 +203,47 @@ _Show = _ShowString >>> _String
 
 _NaturalInteger :: BiMap Natural Integer
 _NaturalInteger = BiMap (Just . toInteger) maybeInteger
-  where maybeInteger :: Integer -> Maybe Natural
-        maybeInteger n
-          | n < 0     = Nothing
-          | otherwise = Just (fromIntegral n)
+  where
+    maybeInteger :: Integer -> Maybe Natural
+    maybeInteger n
+      | n < 0     = Nothing
+      | otherwise = Just (fromIntegral n)
 
 _Natural :: BiMap Natural AnyValue
 _Natural = _NaturalInteger >>> _Integer
 
-_WordInteger :: BiMap Word Integer
-_WordInteger = BiMap (Just . toInteger) maybeInteger
-  where maybeInteger :: Integer -> Maybe Word
-        maybeInteger n
-          | n < toInteger (minBound :: Word) = Nothing
-          | n > toInteger (maxBound :: Word) = Nothing
-          | otherwise                        = Just (fromIntegral n)
+_BoundedInteger :: (Integral a, Bounded a) => BiMap a Integer
```

Comment:
That's wonderful! So it can be used to create a similar function for `Int32` or any other type that you could need in your code! I like this solution 👍

Reply:
I like it too, much cleaner than a copy/paste function for each possible type ^^